packages/core/agent/src/plugins/epoch-monitor/epoch-monitor.ts

```diff
@@ -110,4 +74,49 @@ export class EpochMonitor extends AbstractPlugin {
     this._spaceStates.forEach((state) => state.subscriptions.forEach((subscription) => subscription.unsubscribe()));
     this._spaceStates.clear();
   }
+
+  // TODO(burdon): Subscribe to space members to update address book.
+  private async _monitorSpace(space: Space, state: SpaceState) {
+    state.subscriptions.push(
+      space.members.subscribe((members) =>
+        log('updated', { members: members.map((member) => member.identity.identityKey) }),
+      ),
+    );
+
+    // Monitor spaces owned by this agent.
+    if (this._client!.halo.identity.get()!.identityKey.equals(space.internal.data.creator!)) {
+      log('monitoring', { space: state.spaceKey });
+
+      // Listen for epochs.
+      const limit = this._options.limit ?? DEFAULT_EPOCH_LIMIT;
+      state.subscriptions.push(
+        space.pipeline.subscribe(async (pipeline) => {
+          // TODO(burdon): Rather than total messages, implement inequality in timeframe?
```

Comment:
fyi there's also a `timeframe.newMessages(since)` - how many new messages we're created compared to `since` timeframe